src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/IdentityProviderEntityBuilder.java

```diff
@@ -0,0 +1,106 @@
+                buildIdentityProviderEntity(identityProviderEntry.getKey(), identityProviderEntry.getValue())
+        ).collect(Collectors.toList());
+    }
+
+    private Entity buildIdentityProviderEntity(String name, IdentityProvider identityProvider) {
+        final Element identityProviderElement = document.createElement("l7:IdentityProvider");
+        final String id = idGenerator.generate();
+        identityProviderElement.setAttribute("id", id);
+        identityProviderElement.appendChild(DocumentTools.createElement(document, "l7:Name", name));
+        identityProviderElement.appendChild(DocumentTools.createElement(document, "l7:IdentityProviderType", identityProvider.getType().getValue()));
+
+        switch(identityProvider.getType()) {
+            case BIND_ONLY_LDAP:
+                if (identityProvider.getProperties() != null) {
+                    identityProviderElement.appendChild(BuilderUtils.buildPropertiesElement(
+                            identityProvider.getProperties().entrySet()
+                                    .stream()
+                                    .collect(Collectors.toMap(stringStringEntry -> "property." + stringStringEntry.getKey(), Map.Entry::getValue)), document));
+                }
+                identityProviderElement.appendChild(buildBindOnlyLdapIPDetails(identityProvider));
+                break;
+            case LDAP:
+            case INTERNAL:
+            case FEDERATED:
+            case POLICY_BACKED:
+            default:
+                throw new EntityBuilderException("Please Specify the Identity Provider Type as one of: 'BIND_ONLY_LDAP'");
+        }
+
+        return new Entity("ID_PROVIDER_CONFIG", name, id, identityProviderElement);
+    }
+
+    private Element buildBindOnlyLdapIPDetails(IdentityProvider identityProvider) {
+        final BindOnlyLdapIdentityProviderDetail identityProviderDetail = (BindOnlyLdapIdentityProviderDetail) identityProvider.getIdentityProviderDetail();
+        if (identityProviderDetail == null) {
+            throw new EntityBuilderException("Identity Provider Detail must be specified for BIND_ONLY_LDAP");
+        }
+        final Element extensionElement = document.createElement("l7:Extension");
+        final Element bindOnlyLdapIdentityProviderDetailElement = document.createElement("l7:BindOnlyLdapIdentityProviderDetail");
+        extensionElement.appendChild(bindOnlyLdapIdentityProviderDetailElement);
+        final Element serverUrlsElement = document.createElement("l7:ServerUrls");
+        bindOnlyLdapIdentityProviderDetailElement.appendChild(serverUrlsElement);
+        final List<String> serverUrls = identityProviderDetail.getServerUrls();
+        if (serverUrls == null || serverUrls.isEmpty()) {
+            throw new EntityBuilderException("serverUrls must be a list of urls.");
+        }
+        identityProviderDetail.getServerUrls().forEach(url -> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove useless curly braces around statement ([squid:S1602](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1602))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=58&issues=AWVKO96liZQY_aXE1Ls5&open=AWVKO96liZQY_aXE1Ls5)

[ISSUE_KEY]: AWVKO96liZQY_aXE1Ls5